clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithoutNonRetriableException() {
```

Comment:
@umesh9794 I think Jason is saying that the following line must return a Future that is not complete for this test to verify the fix:

```java
client.poll(future, remainingMs);
```

If that method returns a completed `Future`, then the test will pass without the fix in this PR. You can verify this by removing the fix and running the test. Does it pass, in that case?